common/src/main/java/com/revenuecat/purchases/common/telemetry/TelemetryManager.kt

```diff
@@ -0,0 +1,53 @@
+package com.revenuecat.purchases.common.telemetry
+
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.verboseLog
+
+class TelemetryManager(
+    private val telemetryFileHelper: TelemetryFileHelper,
+    private val telemetryAnonymizer: TelemetryAnonymizer,
+    private val backend: Backend,
+    private val telemetryDispatcher: Dispatcher,
+    private val telemetryEnabled: Boolean
+) {
+    fun syncTelemetryFileIfNeeded() {
+        if (!telemetryEnabled) {
+            verboseLog("Telemetry disabled. Skipping syncing events.")
+            return
+        }
+        telemetryDispatcher.enqueue(
+            command = {
+                if (telemetryFileHelper.telemetryFileIsEmpty()) return@enqueue
+                val telemetryList = telemetryFileHelper.readTelemetryFile()
+                val telemetryCount = telemetryList.size
+                backend.postTelemetry(
+                    telemetryList = telemetryList,
+                    onSuccessHandler = {
+                        verboseLog("Synced telemetry file successfully.")
+                        telemetryFileHelper.cleanSentTelemetry(telemetryCount)
+                    },
+                    onErrorHandler = { error ->
+                        verboseLog("Error syncing telemetry file: $error")
+                        telemetryFileHelper.deleteTelemetryFile()
+                    }
+                )
+            }
+        )
+    }
+
+    fun trackEvent(telemetryEvent: TelemetryEvent) {
+        if (!telemetryEnabled) {
```

Comment:
Small change, but I would take this change out, so this "manager" is only initialized when it's enabled, therefore making it a bit simpler.